### PR TITLE
Improve undefined/null handling for updates

### PR DESCRIPTION
**Pull request by nsams**
- Add IsUndefinable and IsNull (similar, but more specific than IsOptional from class-validator)
- Add custom PartialType (similar to @nestjs/mapped-types but uses IsUndefinable instead of IsOptional)

Update api crud generator to allow partial update in update mutation:
- null: set to null (eg. a relation or a Date)
- undefined: do not touch this field

### CODE REVIEW

File: packages/api/cms-api/src/common/helper/partial-type.helper.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { Type } from "@nestjs/common";
+import { inheritPropertyInitializers, inheritTransformationMetadata, inheritValidationMetadata, MappedType } from "@nestjs/mapped-types";
+
+import { IsUndefinable } from "../validators/is-undefinable";
+
+//Copy from @nestjs/mapped-types with applyIsOptionalDecorator changed to use IsUndefinable instead of IsOptional
+export function PartialType<T>(classRef: Type<T>): MappedType<Partial<T>> {
```
Comment (nsams) [inline packages/api/cms-api/src/common/helper/partial-type.helper.ts:7]: probably not a good idea to use the exact same name as nestjs does, but I couldn't come up with a good name.

Comment (johnnyomair) [reply packages/api/cms-api/src/common/helper/partial-type.helper.ts]: Two suggestions
- PartialUpdateInputType
- PartialUpdateType

**Review by johnnyomair — APPROVED** [review]

File: packages/api/cms-api/src/common/helper/partial-type.helper.ts (hunk 1 above)
Comment (johnnyomair) [reply packages/api/cms-api/src/common/helper/partial-type.helper.ts]: But I'm okay with PartialType too

Comment (fraxachun) [reply packages/api/cms-api/src/common/helper/partial-type.helper.ts]: I'd use PartialType as this does not more or less as described in https://docs.nestjs.com/openapi/mapped-types but better and customized for our needs.

**Pull request merged**